include/dlaf_c/utils.h

```diff
@@ -13,9 +13,11 @@
 /// @file
 
 #ifdef __cplusplus
-#define DLAF_EXTERN_C extern "C"
+#define DLAF_EXTERN_C     extern "C"
+#define DLAF_NOEXCEPT_CPP noexcept
```

Comment:
Since you asked for it... ;) I think the `CPP` is a bit redundant, what about simply using `DLAF_NOEXCEPT`? I put the `C` in `DLAF_EXTERN_C` because it's part of the definition: `extern "C"` -> `DLAF_EXTERN_C`, `noexcept` -> `DLAF_NOEXCEPT`.

But I don't really mind TBH.

Reply:
I did indeed!

Yeah, I wasn't quite happy with the name, but I wanted _some_ suffix since it doesn't unconditionally expand to `noexcept`. `_CPP` was the first thing that came to mind since it applies in C++ mode. Then again, we're never going to have an unconditional `DLAF_NOEXCEPT` on its own since that should just be `noexcept` directly, so I could live with plain `DLAF_NOEXCEPT`.

On a related note, perhaps these should be `DLAF_INTERNAL_*` as well since they're not meant for external consumption?

Reply:
> we're never going to have an unconditional `DLAF_NOEXCEPT` on its own since that should just be `noexcept` directly

Yes, this was basically my thinking, that's why I feel `_CPP` is redundant. But as I said, it does not bother me.

Reply:
Note, I haven't renamed anything yet (i.e. it's still `DLAF_NOEXCEPT_CPP`). Looking for opinions from others still.

Reply:
@rasolca and @albestro agree to changing it to `DLAF_NOEXCEPT`. Will update.

Reply:
Updated to `DLAF_NOEXCEPT`.